qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains functions to construct many-body observables whose expectation
+values can be used to simulate molecular properties.
+"""
+# pylint: disable=too-many-arguments, too-few-public-methods
+import os
+import numpy as np
+from openfermion.hamiltonians import MolecularData
+from openfermion.ops import FermionOperator
+from openfermion.transforms import bravyi_kitaev, jordan_wigner
+
+from . import structure
+
+
+def s2_me_table(sz, n_spin_orbs):
```

Comment:
I'm trying to think of a function name that better communicates to users/developers what this function does. `table` is probably redundant, we should instead emphasize the 'two particle spin' and 'matrix elements' part.

* `spin_matrix_elements`

* `spin2_matrix_elements`

* `two_particle_spin_elements`

Any other thoughts? 🤔

Reply:
Another option is just `spin2_matrix`.

Reply:
I think the simpler way would be to call it ``spin2_coeffs``. Matrix elements may be confusing as you can also compute matrix elements of the full operator, indeed, this is what it is mostly done.